app/Services/EdgeProxyRemoteRouteService.php

```
@@ -0,0 +1,1562 @@
+
+    private function resolveComposeServicePorts(array $compose, string $serviceName): ?array
+    {
+        $serviceConfig = $this->resolveComposeServiceConfig($compose, $serviceName);
+        if (! is_array($serviceConfig)) {
+            return null;
+        }
+
+        $ports = data_get($serviceConfig, 'ports');
+
+        return is_array($ports) ? $ports : null;
+    }
```

PHPMD (2.15.0)

[warning] Avoid unused private methods such as 'resolveComposeServicePorts'. (undefined)

(UnusedPrivateMethod)